### ISSUE TITLE
Datatype mismatch between input labels and `predicted_label`  returned by `CrossValCurate.fit_transform`

### ISSUE THREAD

**Issue by sumanthprabhu** [sumanthprabhu/DQC-Toolkit#8]
**Describe the bug**
The `predicted_label` returned by `CrossValCurate.fit_transform` seems to be of type 'int' even when the input label contains text values. Results become uninterpretable since`predicted_label` and input labels have zero overlap.

**To Reproduce**
```py
from datasets import load_dataset
from dqc import CrossValCurate
import pandas as pd 

dataset = 'SetFit/20_newsgroups'
dset = load_dataset(dataset)
train_data = pd.DataFrame(dset['train'])

cvc = CrossValCurate()
train_data_modified = cvc.fit_transform(train_data, y_col_name='label_text')
print(type(train_data_modified['predicted_label'].values[0]) == type(train_data['label_text'].values[0]))
```

**Expected behavior**
True (both contain text values)

**Actual results**
False (input labels are text values while predictions are integers)

**System Information:**
 'os_type': 'Linux',
 'os_version': '5.15.133+',
 'python_version': '3.10.13',
 'dqc-toolkit': '0.1.1',
 'transformers': '4.39.3',
 'sentence_transformers': '2.7.0',
 'scikit-learn': '1.5.0'

**Additional context**
